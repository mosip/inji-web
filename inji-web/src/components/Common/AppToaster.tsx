import React from "react";
import {ToastContainer} from "react-toastify";

export const AppToaster: React.FC = () => {
    return <ToastContainer
        position="top-right"
        autoClose={1000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        icon={<React.Fragment/>}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        style={{width: '400px'}}
        theme="colored"
    />
}
