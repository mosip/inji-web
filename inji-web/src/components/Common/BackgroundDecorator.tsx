import React from 'react';

interface BackgroundDecoratorProps {
    circleCount?: number;
    topOffset?: number;
    baseRadius?: number;
    logoSrc?: string;
    logoAlt?: string;
    testId?: string;
}

export const BackgroundDecorator: React.FC<BackgroundDecoratorProps> = ({
    circleCount = 6,
    topOffset = 155,
    baseRadius = 96,
    logoSrc,
    logoAlt = "Logo",
    testId = "background-decorator"
}) => {
    return (
        <div 
            className="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
            data-testid={testId}
        >
            <div className={`absolute top-[${topOffset}px]`} style={{ top: `${topOffset}px` }}>
                {[...Array(circleCount)].map((_, index) => {
                    const radius = baseRadius + index * baseRadius;
                    const opacity = 0.8 - index * 0.1;
                    return (
                        <div
                            key={index}
                            className="absolute rounded-full border overflow-hidden"
                            style={{
                                width: `${radius}px`,
                                height: `${radius}px`,
                                borderWidth: '1px',
                                borderColor: `rgba(228, 231, 236, ${opacity})`,
                                top: `calc(50% - ${radius / 2}px)`,
                                left: `calc(50% - ${radius / 2}px)`
                            }}
                            data-testid={`${testId}-circle-${index}`}
                        />
                    );
                })}
                
                {logoSrc && (
                    <div
                        className="flex items-center justify-center"
                        data-testid={`logo-${testId}-container`}
                    >
                        <img
                            src={logoSrc}
                            alt={logoAlt}
                            data-testid={`logo-${testId}`}
                        />
                    </div>
                )}
            </div>
        </div>
    );
};