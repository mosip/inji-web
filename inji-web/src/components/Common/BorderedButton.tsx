import React from "react";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={"bg-gradient-to-r rounded-lg from-iw-primary to-iw-secondary p-0.5"}>
            <div className="py-2 px-4 rounded-lg bg-white">
                <button
                    data-testid={props.testId}
                    onClick={props.onClick}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title}
                </button>
            </div>
    </div>
}

export type BorderedButtonProps = {
    testId: string;
    onClick: ()=> void;
    title: string;
}
