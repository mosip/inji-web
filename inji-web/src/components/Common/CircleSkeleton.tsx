type CircleSkeletonProps = {
    size?: string;
    className?: string;
};

export const CircleSkeleton = ({
                                   size = 'w-20 h-20',
                                   className = '',
                               }: CircleSkeletonProps) => {
    return (
        <div
            className={`rounded-full bg-gray-300 animate-pulse ${size} ${className}`}
        />
    );
};
  