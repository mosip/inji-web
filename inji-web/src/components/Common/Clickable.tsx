import React from 'react';

export interface ClickableProps {
    onClick: () => void;
    children: React.ReactNode;
    testId: string;
    className: string;
}

export const Clickable: React.FC<ClickableProps> = ({
                                                        onClick,
                                                        children,
                                                        testId,
                                                        className = '',
                                                    }) => {
    const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
        if (event.key === 'Enter' || event.key === ' ') {
            event.preventDefault();
            onClick();
        }
    };

    return (
        <div
            role="menuitem"
            tabIndex={0}
            className={className}
            onClick={onClick}
            onKeyDown={handleKeyPress}
            data-testid={testId}
        >
            {children}
        </div>
    );
};
