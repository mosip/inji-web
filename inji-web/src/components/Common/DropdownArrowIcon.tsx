import React from 'react';
import {RiArrowUpSFill, RiArrowDownSFill} from 'react-icons/ri';
import { DropdownArrowIconProps } from '../../types/data';
import { GradientWrapper } from './GradientWrapper';

const DropdownArrowIcon: React.FC<DropdownArrowIconProps> = ({isOpen, size = 20, color = 'var(--iw-color-languageArrowIcon)'}) => {
    const ArrowIcon = isOpen ? RiArrowUpSFill : RiArrowDownSFill;

    return (
        <GradientWrapper>
            <ArrowIcon size={size} color={color} />
        </GradientWrapper>
    );
};

export default DropdownArrowIcon;
