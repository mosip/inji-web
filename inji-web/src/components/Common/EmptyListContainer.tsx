import React from "react";
import {EmptyListContainerProps} from "../../types/components";

export const EmptyListContainer: React.FC<EmptyListContainerProps> = ({content}) => {
    return <React.Fragment>
        <div data-testid="EmptyList-Outer-Container"
             className="flex justify-center items-center w-full mx-auto flex-col">
            <div data-testid="EmptyList-Inner-Container" className="container mx-auto mt-8 px-4 flex-1 flex flex-col">
                <p data-testid="EmptyList-Text" className="text-center">{content}</p>
            </div>
        </div>
    </React.Fragment>
}
