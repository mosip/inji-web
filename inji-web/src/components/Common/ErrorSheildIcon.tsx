import {BsShieldFillX} from "react-icons/bs";
import React from "react";

export const ErrorSheildIcon: React.FC = () => {
  return <div className="rounded-full p-2 shadow bg-white">
      <div className="rounded-full p-8 bg-iw-shieldErrorShadow">
          <BsShieldFillX
              data-testid="DownloadResult-Error-ShieldIcon" size={40} color={'var(--iw-color-shieldErrorIcon)'}/>
      </div>
  </div>
}
