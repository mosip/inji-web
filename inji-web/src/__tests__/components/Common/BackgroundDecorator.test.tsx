import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {BackgroundDecorator} from '../../../components/Common/BackgroundDecorator';

jest.mock('../../../assets/Logomark.png', () => 'mocked-logo-path');

const renderComponent = (props = {}) => {
    return render(<BackgroundDecorator {...props} />);
};

describe('BackgroundDecorator Component', () => {
    test('matches snapshot with default props', () => {
        const {asFragment} = renderComponent();

        expect(asFragment()).toMatchSnapshot();
    });

    test('matches snapshot with custom props', () => {
        const {asFragment} = renderComponent({
            logoAlt: 'Custom Logo Alt',
            circleCount: 4,
            topOffset: 100,
            baseRadius: 80,
            testId: 'custom-decorator',
            logoTestId: 'custom-logo'
        });

        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the component with default test ID', () => {
        renderComponent();

        expect(screen.getByTestId('background-decorator')).toBeInTheDocument();
    });

    test('renders the component with custom test ID', () => {
        renderComponent({testId: 'custom-decorator'});

        expect(screen.getByTestId('custom-decorator')).toBeInTheDocument();
    });

    test('renders concentric circles based on the provided count value', () => {
        const count = 3;
        const testId = 'circle-test';
        renderComponent({circleCount: count, testId});

        for (let i = 0; i < count; i++) {
            expect(
                screen.getByTestId(`${testId}-circle-${i}`)
            ).toBeInTheDocument();
        }
    });

    test('uses custom topOffset for positioning the circle container', () => {
        const topOffset = 200;
        renderComponent({topOffset});

        const circleContainer = screen
            .getByTestId('background-decorator')
            .querySelector('div');

        expect(circleContainer).toHaveStyle(`top: ${topOffset}px`);
    });

    test('uses custom baseRadius for circles', () => {
        const baseRadius = 50;
        const testId = 'radius-test';
        renderComponent({baseRadius, testId, circleCount: 1});

        const firstCircle = screen.getByTestId(`${testId}-circle-0`);

        expect(firstCircle).toHaveStyle(`width: ${baseRadius}px`);
        expect(firstCircle).toHaveStyle(`height: ${baseRadius}px`);
    });
});
