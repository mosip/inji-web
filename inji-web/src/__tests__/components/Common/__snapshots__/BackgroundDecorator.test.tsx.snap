// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundDecorator Component matches snapshot with custom props 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
    data-testid="custom-decorator"
  >
    <div
      class="absolute top-[100px]"
      style="top: 100px;"
    >
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="custom-decorator-circle-0"
        style="width: 80px; height: 80px; border-width: 1px; border-color: rgba(228, 231, 236, 0.8); top: calc(50% - 40px); left: calc(50% - 40px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="custom-decorator-circle-1"
        style="width: 160px; height: 160px; border-width: 1px; border-color: rgba(228, 231, 236, 0.7000000000000001); top: calc(50% - 80px); left: calc(50% - 80px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="custom-decorator-circle-2"
        style="width: 240px; height: 240px; border-width: 1px; border-color: rgba(228, 231, 236, 0.6000000000000001); top: calc(50% - 120px); left: calc(50% - 120px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="custom-decorator-circle-3"
        style="width: 320px; height: 320px; border-width: 1px; border-color: rgba(228, 231, 236, 0.5); top: calc(50% - 160px); left: calc(50% - 160px);"
      />
      <div
        class="flex items-center justify-center"
        data-testid="logo-inji-web-container"
      >
        <img
          alt="Inji Web Logo"
          data-testid="logo-inji-web"
          src="mocked-logo-path"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BackgroundDecorator Component matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
    data-testid="background-decorator"
  >
    <div
      class="absolute top-[116px]"
      style="top: 116px;"
    >
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="background-decorator-circle-0"
        style="width: 96px; height: 96px; border-width: 1px; border-color: rgba(228, 231, 236, 0.8); top: calc(50% - 48px); left: calc(50% - 48px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="background-decorator-circle-1"
        style="width: 192px; height: 192px; border-width: 1px; border-color: rgba(228, 231, 236, 0.7000000000000001); top: calc(50% - 96px); left: calc(50% - 96px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="background-decorator-circle-2"
        style="width: 288px; height: 288px; border-width: 1px; border-color: rgba(228, 231, 236, 0.6000000000000001); top: calc(50% - 144px); left: calc(50% - 144px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="background-decorator-circle-3"
        style="width: 384px; height: 384px; border-width: 1px; border-color: rgba(228, 231, 236, 0.5); top: calc(50% - 192px); left: calc(50% - 192px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="background-decorator-circle-4"
        style="width: 480px; height: 480px; border-width: 1px; border-color: rgba(228, 231, 236, 0.4); top: calc(50% - 240px); left: calc(50% - 240px);"
      />
      <div
        class="absolute rounded-full border overflow-hidden"
        data-testid="background-decorator-circle-5"
        style="width: 576px; height: 576px; border-width: 1px; border-color: rgba(228, 231, 236, 0.30000000000000004); top: calc(50% - 288px); left: calc(50% - 288px);"
      />
      <div
        class="flex items-center justify-center"
        data-testid="logo-inji-web-container"
      >
        <img
          alt="Inji Web Logo"
          data-testid="logo-inji-web"
          src="mocked-logo-path"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
